alluxio-ui/master/src/containers/App/App.tsx

```diff
@@ -0,0 +1,163 @@
+                <Route path="/overview" exact={true} render={this.renderOverview}/>
+                <Route path="/browse" exact={true} render={this.renderBrowse}/>
+                <Route path="/config" exact={true} component={this.renderConfiguration}/>
+                <Route path="/data" exact={true} component={this.renderData}/>
+                <Route path="/logs" exact={true} component={this.renderLogs}/>
+                <Route path="/metrics" exact={true} component={this.renderMetrics}/>
+                <Route path="/workers" exact={true} component={this.renderWorkers}/>
+                <Route render={this.renderError}/>
+              </Switch>
+            </div>
+            <div className="container-fluid footer-wrapper">
+              <Footer data={footerNavigationData}/>
+            </div>
+          </div>
+        </ConnectedRouter>
+      </Provider>
+    );
+  }
+
+  private redirectToOverview(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Redirect to="/overview"/>
+    );
+  }
+
+  private renderOverview(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Overview {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderBrowse(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Browse {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderConfiguration(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Configuration {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderData(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Data {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderLogs(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Logs {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderMetrics(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Metrics {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderWorkers(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return (
+      <Workers {...routerProps} refreshValue={this.state.refreshValue}/>
+    );
+  }
+
+  private renderError(routerProps: RouteComponentProps<any, StaticContext, any>) {
+    return null;
+  }
+
+  private flipRefreshValue() {
+    this.setState({refreshValue: !this.state.refreshValue});
+  }
+
+  private setAutoRefresh(shouldAutoRefresh: boolean) {
+    if (shouldAutoRefresh && !this.intervalHandle) {
+      this.intervalHandle = setInterval(this.flipRefreshValue, this.refreshInterval);
+    } else {
+      if (this.intervalHandle) {
+        clearInterval(this.intervalHandle);
+        this.intervalHandle = null;
+      }
+    }
+  }
+}
+
+const mapStateToProps = () => ({});
```

Comment:
Should we be passing `refreshValue` as a prop?

Reply:
This maps the store to the local view props, since the timer is bound to the app view (because it has the header component), I use it to propagate from the app view downwards to all the components that care.